### PR TITLE
Instrument async code with tracing spans

### PR DESCRIPTION
**Pull request by rukai**
Concurrent logs now look like this:

image

This allows us to see through the noise and form a sequence of events from concurrent logs.

Possibly we should be using a random id instead of an incrementing value. - Discussed with kuangda: incrementing will be fine for now

Relevant docs: https://docs.rs/tracing/0.1.29/tracing/trait.Instrument.html

### CODE REVIEW

File: shotover-proxy/src/transforms/chain.rs [hunk 1]
```diff
@@ -115,43 +115,48 @@ impl TransformChain {
...
 
         let mut chain = self.clone();
-        let _jh = tokio::spawn(async move {
-            while let Some(ChannelMessage {
-                return_chan,
-                messages,
-            }) = rx.recv().await
-            {
-                #[cfg(test)]
+        let _jh = tokio::spawn(
```
Comment (rukai) [inline shotover-proxy/src/transforms/chain.rs:118]: Very little was actually change here, the only thing I did here was call .in_current_span() but it made rust fmt freak out.

**Review by conorbros — CHANGES_REQUESTED** [review]
Everything looks good to me but we should add some tests before merging. 

Testing tracing: https://docs.rs/tracing-test/0.1.0/tracing_test/

**Comment by rukai** [comment]
That library is way too inflexible to do anything with:
* No way to assert on spans (What this PR is actually changing)
* No way to assert on part of a log. e.g. we cant write an assertion for `New connection from 127.0.0.1:38602` because the port is random.
* No way to view the actual log after a failure.

The idea is cool but I think we'll have to revisit that later when its more than just a prototype.

**Review by conorbros — APPROVED** [review]
Fair enough. Would be nice to test but should be fine for the time being.

**Review by XA21X — APPROVED** [review]

**Pull request merged**